src/docs/stan-reference/examples.tex

```diff
@@ -8046,6 +8046,154 @@ \section{Circular Representations of Days and Years}
+  real dat_r[0];
+  int dar_i[0];
+}
+
+transformed parameters {
+  real theta[2] = {3, 6};
+  real x[2];
+
+  x = algebra_solver(guess, theta, dat_r, dat_i);
+}
+\end{stancode}
+
+which returns $x = \{3, -2\}$.
+
+\subsection{Data versus Parameters}
+The arguments for the initial guess \code{guess}, the real data \code{dat\_r}, and
+the integer data \code{dat\_i} must be expressions that only involve data or
+transformed data variables. The parameters \code{theta} is the only argument
+which may involve parameters.
+
+\subsection{Length of the Algebraic Function and of the Vector of Unknowns}
+The Jacobian of the solution with respect to the parameters is computed
+using the implicit function theorem, which places certain restrictions. In particular,
+the Jacobian of the algebraic function $f$ with respect to the unknowns $x$ must
+be invertible. This requires the Jacobian to be square, meaning \textit{$f(x)$ and
+$x$ have the same length} or, in other words \textit{the number of equations in
+the system is the same as the number of unknowns.}
+
+\subsection{Pathological Solutions}
+Certain systems may be degenerate, meaning they have multiple solutions. The 
+algebraic solver will not report these cases, as the algorithm stops once it has found
+an acceptable solution. The initial guess will often determine which solution gets found
+first. The degeneracy may be broken by putting additional constraints on the solution.
+For instance, it might make physical sense for a solution to be positive or negative.
+
+On the other hand, a system may not have a solution (for a given point in the parameter
+space). In that case, the solver will not converge to a solution. When the solver fails to
+do so, the current metropolis proposal gets rejected.
+
+\section{Control Parameters for the Algebraic Solver}
+%
+The call to the algebraic solver shown above uses the default control settings. The solver
+allows three additional parameters, all of which must be supplied if any of them is
+required.
+%
+\begin{stancode}
+  x = algebra_solver(guess, theta, dat_r, dat_i, 
+                     rel_tol, f_tol, max_steps);
+\end{stancode}
+
+The three control arguments are relative tolerance, function tolerance, and maximum
+number of steps. Their default values are respectively \code{1e-10} ($10^{-10}$), \code{1e-6}
+($10^{-6}$), and \code{1e3} ($10^3$).
+
+\subsection{Tolerance}
+%
+The relative and function tolerances control the accuracy of the solution generated by
+the solver. Relative tolerances are relative to the solution value. The function tolerance
+is the norm of the algebraic function, once we plug in the proposed solution. This norm
+should go to 0 (equivalently, all elements of the vector function are 0). It helps to think about this
+geometrically. Ideally the output of the algebraic function is at the origin; the norm measures
+deviations from this ideal. Users should remember the norm scales with the square root of
+$f$'s length.
```

Comment:
The norm we get from Pythagoras' theorem. An alternative would have been to use the norm for individual elements of z. That is require || z[i] || < ftol, for all i. If there are n elements in the vector, this constraint very roughly translates to || z || < ftol * sqrt(n). This what the thought I was trying to express. For clarity, I'll simply state:

> As the length of z increases, a certain function tolerance becomes an increasingly difficult criteria to meet, since the deviations from every individual element of the vector add up.